### ISSUE TITLE
simplify-boolean-expression is not correct for ['1', ''][x]

### ISSUE THREAD

**Issue by sk-** [PyCQA/pylint#2473]
### Steps to reproduce
1. Run pylint on the code
```python
def foo(x):
  return ['1', ''][x]
```

### Current behavior
the error:

```
R:  2, 2: Boolean expression may be simplified to '1' (simplify-boolean-expression)
```

Note that if we change the array to either `['1', '2']` or `['', `2`]`, then the error is:

```
Consider using ternary ('2' if x else '') (consider-using-ternary)
```

### Expected behavior
No warning (or consider-using-ternary) should be raised, as it is not possible to simplify the code.

### pylint --version output
```
pylint 2.1.1
astroid 2.0.4
Python 3.6.5 (v3.6.5:f59c0932b4, Mar 28 2018, 03:03:55)
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)]
```

**Comment by PCManticore** [comment]
Thank you! This should not happen.

**Comment by PCManticore** [comment]
Thanks for reporting! decided to remove the sequence based check for potential ternary operations, as it seems more of an heuristic than something we can actually infer correctly as a ternary operation in disguise.
